Make rollup an optional peer dependency

Currently the plugin-check and plugin-worker packages relies on a couple Rollup plugins that are used in the Vite config (at runtime, meaning during the `sde bundle/dev` process).  These don't actually need rollup when used via vite, but they still list rollup as a required peer dependency, which causes issues when installing via pnpm (see below).

```
hello-world campbell$ pnpm i
 WARN  deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
Already up-to-date
 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

.
└─┬ @sdeverywhere/plugin-check
  ├─┬ @rollup/plugin-node-resolve
  │ ├── ✕ missing peer rollup@^2.42.0
  │ └─┬ @rollup/pluginutils
  │   └── ✕ missing peer rollup@^1.20.0||^2.0.0
  └─┬ @rollup/plugin-replace
    └── ✕ missing peer rollup@"^1.20.0 || ^2.0.0"
Peer dependencies that should be installed:
  rollup@">=2.42.0 <3.0.0"  
```

We may be able to avoid this by marking rollup as optional using [peerDependenciesMeta](https://pnpm.io/package_json#peerdependenciesmetaoptional).

--- a/packages/plugin-check/package.json
+++ b/packages/plugin-check/package.json
@@ -41,8 +41,18 @@
     "@sdeverywhere/runtime-async": "^0.1.0",
     "assert-never": "^1.2.1",
     "picocolors": "^1.0.0",
+    "rollup": "^2.76.0",
     "vite": "^2.9.12",
     "vite-plugin-glob": "^0.3.2"
+  },
+  "dependenciesComments": {
+    "rollup": [
+      "XXX: For now, rollup is listed as a dependency even though it is not actually",
+      "needed since we only use the @rollup plugins with vite.  Those @rollup plugins",
+      "don't technically need rollup to function properly, but they list rollup as a",
+      "required peer dependency.  To avoid making consumers worry about this, we list",
+      "it as a dependency here."
+    ]
   },
   "devDependencies": {
     "@types/node": "^16.11.7"

--- a/packages/plugin-worker/package.json
+++ b/packages/plugin-worker/package.json
@@ -34,7 +34,17 @@
     "@sdeverywhere/build": "^0.1.0",
     "@sdeverywhere/runtime": "^0.1.0",
     "@sdeverywhere/runtime-async": "^0.1.0",
+    "rollup": "^2.76.0",
     "vite": "2.9.6"
+  },
+  "dependenciesComments": {
+    "rollup": [
+      "XXX: For now, rollup is listed as a dependency even though it is not actually",
+      "needed since we only use the @rollup plugins with vite.  Those @rollup plugins",
+      "don't technically need rollup to function properly, but they list rollup as a",
+      "required peer dependency.  To avoid making consumers worry about this, we list",
+      "it as a dependency here."
+    ]
   },
   "devDependencies": {
     "@types/node": "^16.11.7"

